app/models/course/registration.rb

```diff
@@ -0,0 +1,29 @@
+class Course::Registration < ActiveRecord::Base
+  stampable
+
+  scope :staff, -> { where(role: roles.values_at(:teaching_assistant, :manager, :owner)) }
```

Comment:
What about this?
https://gist.github.com/26c7aebbfeb6d7f9328e.git
Well, we can also do the simpler way. The hash returned by `CourseUser.roles` is just a normal hash where we can add anything. So we can even just manually add those extra things to the roles. However I can foresee that methods like `CourseUser.role_scopes` and `course_user.staff?` may be useful.